src/json/json.cc

```diff
@@ -0,0 +1,3232 @@
+    VALKEYMODULE_NOT_USED(privdata);
+    return keyTable->getNumShards();
+}
+
+int Config_SetKeyTableNumShards(const char *name, long long val, void *privdata, ValkeyModuleString **err) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    VALKEYMODULE_NOT_USED(err);
+    return handleSetNumShards(&val);
+}
+
+long long Config_GetHashTableGrowFactor(const char *name, void *privdata) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    return rapidjson::hashTableFactors.grow * 100;
+}
+
+int Config_SetHashTableGrowFactor(const char *name, long long val, void *privdata, ValkeyModuleString **err) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    VALKEYMODULE_NOT_USED(err);
+    return handleHashTableFactor(&rapidjson::HashTableFactors::grow, &val, 100.f);
+}
+
+long long Config_GetHashTableShrinkFactor(const char *name, void *privdata) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    return rapidjson::hashTableFactors.shrink * 100;
+}
+
+int Config_SetHashTableShrinkFactor(const char *name, long long val, void *privdata, ValkeyModuleString **err) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    VALKEYMODULE_NOT_USED(err);
+    return handleHashTableFactor(&rapidjson::HashTableFactors::shrink, &val, 100.f);
+}
+
+long long Config_GetHashTableMinLoadFactor(const char *name, void *privdata) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    return rapidjson::hashTableFactors.minLoad * 100;
+}
+
+int Config_SetHashTableMinLoadFactor(const char *name, long long val, void *privdata, ValkeyModuleString **err) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    VALKEYMODULE_NOT_USED(err);
+    return handleHashTableFactor(&rapidjson::HashTableFactors::minLoad, &val, 100.f);
+}
+
+long long Config_GetHashTableMaxLoadFactor(const char *name, void *privdata) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    return rapidjson::hashTableFactors.maxLoad * 100;
+}
+
+int Config_SetHashTableMaxLoadFactor(const char *name, long long val, void *privdata, ValkeyModuleString **err) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    VALKEYMODULE_NOT_USED(err);
+    return handleHashTableFactor(&rapidjson::HashTableFactors::maxLoad, &val, 100.f);
+}
+
+long long Config_GetHashTableMinSize(const char *name, void *privdata) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    return rapidjson::hashTableFactors.minHTSize;
+}
+
+int Config_SetHashTableMinSize(const char *name, long long val, void *privdata, ValkeyModuleString **err) {
+    VALKEYMODULE_NOT_USED(name);
+    VALKEYMODULE_NOT_USED(privdata);
+    VALKEYMODULE_NOT_USED(err);
+    return handleHashTableFactor(&rapidjson::HashTableFactors::minHTSize, &val, size_t(1));
+}
+
+int registerModuleConfigs(ValkeyModuleCtx *ctx) {
```

Comment:
The RFC has 2 configs, why does this have like 50?

Reply:
5️⃣0️⃣⁉️

It's easy to add configs but it's a breaking change to remove one. Without looking at what they are for, can we replace them with constants or something before we release a version where we commit to maintaining them?

Reply:
We considered dropping these configs. However, it requires a decent amount of code cleanup. Agreed we should remove them. We can create an issue to replace these configs with constants.

Reply:
https://github.com/valkey-io/valkeyJSON/issues/2